src/test/java/org/jabref/logic/importer/util/GrobidServiceTest.java

```diff
@@ -72,6 +73,11 @@ public void processInvalidCitationTest() {
         assertThrows(IOException.class, () -> grobidService.processCitation("iiiiiiiiiiiiiiiiiiiiiiii", importFormatPreferences, GrobidService.ConsolidateCitations.WITH_METADATA));
     }
 
+    @Test
+    public void failsWhenGrobidDisabled() {
+        assertThrows(UnsupportedOperationException.class, () -> new GrobidService(new ImportSettingsPreferences(false, false, false, "http://grobid.jabref.org:8070")));
```

Comment:
Maybe extract the ImportSettingsPreferences (I hate that name) to a separate variable, following the typical test scheme:
```
// given
...
// when
...
// then
...
```

See https://blog.codecentric.de/en/2017/09/given-when-then-in-junit-tests/ for example

Reply:
Extracting is easy and I agree it is more readable. For the 'when' clause, since I expect an exception, something like [this](https://stackoverflow.com/a/33933067/5401297) would be necessary. In my opition expectThrows is easier to read though. Is it ok to violate given-when-then in this case?

Reply:
This is not a hard law, but a suggestion. Should make reading and understanding tests easier. If in this case it's not, then ok.